bin/express-cli.js

```diff
@@ -160,7 +166,9 @@ function createApplication (name, dir, options, done) {
 
   // copy route templates
   mkdir(dir, 'routes')
-  copyTemplateMulti('js/routes', dir + '/routes', '*.js')
+  copyTemplateMulti(
+    options.es6 ? 'mjs/routes' : 'js/routes',
+    dir + '/routes', '*.js')
```

Comment:
Having completely separate files/templates for the ES6 style is easier for me to understand quickly. What are you thoughts about using `.mjs` file extension and creating a different `.mjs` template? That would remove the need for conditional logic in the `app.js.ejs` file.

Reply:
> Having completely separate files/templates for the ES6 style is easier for me to understand quickly. What are you thoughts about using `.mjs` file extension and creating a different `.mjs` template? That would remove the need for conditional logic in the `app.js.ejs` file.

Actually, I had it that way on an earlier commit. But I thought that having to maintain parallel files might be a problem, so I went with the merged version. If you prefer parallel files, I'll change back.

Reply:
I assumed that the benefit for having separate files or even a separate folder of files only for the es6 templates would make the code easier to understand. "easier" here being "less conditionals to read in an ejs file without the benefit for code formatting helping the eyes follow the code".

What were your observations related to code readability when you had them separate?

Reply:
I'd say that it was definitely more readable. The only issue would be, anyone maintaining this would have to remember to make any changes to app.js.ejs in two places, and there would be greater chance of introducing errors.
Also, what about www? I had that in two places originally.

Reply:
So the tradeoff would be improving code readability vs reducing Express app template maintenance effort. Both of these are for the future developer (which could be any of us ;)

I'm guessing there would have to be an API change in Express to cause a need to change the templates. That possibility might be pretty high given that development activity on Express is going up right now.

On the other hand, this feature (--es6) is essentially a different way to write an Express app. Other than the module loading syntax, the differences between the 2 styles can be opaque, but are there none the less. I would argue that having 2 different sets of templates increases the visibility of the design, which in theory, could reduce maintenance effort.

Ultimately, it's your call. I just wanted to put my 2 cents out there.

Reply:
My latest push changes to parallel versions of app and www.